### PR TITLE
Slideshow: Use RenderContextGl for LayerRendererGl

### PR DESCRIPTION
**Pull request by codewithvk**
Refactored LayerRendererGl to utilize RenderContextGl, replacing repeated functions to reduce code duplication and improve maintainability.

Change-Id: I929ca8bb65897e98a10b4efa92c068a257cd9e16

* Resolves: # 
* Target version: master

### CODE REVIEW

File: browser/src/slideshow/LayerRenderer.ts [hunk 1]
```diff
@@ -47,10 +48,8 @@ class LayerRendererGl implements LayerRenderer {
 
 	constructor(offscreenCanvas: OffscreenCanvas) {
 		this.offscreenCanvas = offscreenCanvas;
-		this.gl = this.offscreenCanvas.getContext('webgl');
-		if (!this.gl) {
-			throw new Error('WebGL not supported');
```
Comment (eszkadev) [inline browser/src/slideshow/LayerRenderer.ts:52]: please leave this exception, I think we need that for detecting when to use 2d fallback

Comment (codewithvk) [reply browser/src/slideshow/LayerRenderer.ts:52]: Nice point

Comment (codewithvk) [reply browser/src/slideshow/LayerRenderer.ts:52]: @eszkadev I have tested, It works fine without it as we already throwing error from `RenderContextGl` constructor so we don't need to worry about it :)

**Review by mcecchetti — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**